### ISSUE TITLE
expo-av doesn't properly do background playback in Android

### ISSUE THREAD

**Issue by mihaibulic2** [expo/expo#26216]
### Summary

expo-av claims it can do background playback in Android, but it doesn't use a foreground service (no notification is shown to the user), so it's actually a best effort sort of thing where it's likely the OS will just kill our app and stop playback after a bit (esp if Doze kicks in).

To do this properly, expo-av needs to run a foreground service and show the user a notification. If we as the dev need to do anything to enable this, it should be documented.

At the very least, the documentation should be updated to warn that this isn't real background playback.

### Link to the related docs page

https://docs.expo.dev/versions/latest/sdk/audio/#playing-or-recording-audio-in-background

**Comment by joranb** [comment]
Yes, really annoying that one. I added a feature request on a related manner here: https://expo.canny.io/feature-requests/p/possibility-to-change-audio-in-background-android

The background playback support on Android is unfortunately not as good as on iOS.

**Comment by nkabram** [comment]
Android kills the app after 60 seconds if it's running in the background. It needs a foreground service to work; I agree that they should at least update the documentation to specify limits on how long expo-av will run in the background. This module is not helpful with this limitation.

**Comment by vijaystroup** [comment]
Upgraded to SDK 51 from 49 and expo-av 14.0.4 from 13.4.1.
Background audio recording still working for iOS but no longer works when I lock the phone for Android.

**Comment by joranb** [comment]
@vijaystroup Did you remember to add the WAKE_LOCK permission to the android permissions part of app.config.js/ts?

Had the same issue before I added that, and Expo did not mention it in their documentation for expo-av, so was a pain to figure out.

**Comment by vijaystroup** [comment]
> @vijaystroup Did you remember to add the WAKE_LOCK permission to the android permissions part of app.config.js/ts?
> 
> Had the same issue before I added that, and Expo did not mention it in their documentation for expo-av, so was a pain to figure out.

Hey @joranb, I just tried that:
app.json
```json
{
  "expo": {
    "android": {
      "permissions": ["android.permission.WAKE_LOCK"]
    }
  }
}
```
I then read in the config here into my app.config.ts.

Still the same issue.

**Comment by joranb** [comment]
@vijaystroup too bad, not sure if it matters, but we have only "WAKE_LOCK", not "android.permission.WAKE_LOCK".

**Comment by vijaystroup** [comment]
> @vijaystroup too bad, not sure if it matters, but we have only "WAKE_LOCK", not "android.permission.WAKE_LOCK".

No luck with that either unfortunately.

**Comment by askpatrickw** [comment]
I'm wondering, is there any update on this for Android?

If someone can provide guidance, my team is willing to help add this feature.

**Comment by ahbou** [comment]
As @mihaibulic2 stated without a running a foreground service, the OS will kill your app.

**Comment by George3d6** [comment]
Curious about this too - `expo-av` seems to be missing a lot of basic functionality or have it poorly documented and background play is one such thing.

**Comment by vijaystroup** [comment]
There is also a new `expo-audio`
https://docs.expo.dev/versions/latest/sdk/audio/

**Comment by NicolasDevJS** [comment]
Hi guys, I created an Android and IOD app using expo av and I want to make the controls just play and pause in the background, is this not possible?

**Comment by joranb** [comment]
@NicolasDevJS Not by using expo-av, but if it is video you are using it for the new expo-video library do support that. As mentioned earlier in this thread there is a new expo-audio library on the way (it is currently in alpha/beta or something) for audio, but I have not seen any confirmation for support of background playback via lock controls like expo-video. But it would surprise me if they did not reuse the way they implemented it for expo-video, fingers crossed. Then the expo audio and video library would handle most use cases, and no need for custom native code for something as trivial as background playback.
